ImageReciver.cs

```diff
@@ -0,0 +1,38 @@
+using UnityEngine;
+using System;
+using System.IO;
+using System.Text;
+using Newtonsoft.Json.Linq;
+using LLMUnity;
+using LLMUnitySamples;
+using TMPro;
+public class ImageReciver : MonoBehaviour
+{
+    //Noskop_dev was here :D
+
+    //This field is used to relay the image to the AI, this can be done by both a URL or a file in your system.
+   
+    public TextMeshProUGUI AnyImageData;
+
+    // Should work with any script that calls the Chat function on the LLMCharacter script. 
+    public AndroidDemo AD;
+  
+  
+
+    void Start()
+    {
+      
+       
+      
+        
+    }
+
+   
+    
+    public void SendImageToAI()  
+    {
+        AD.onInputFieldSubmit(" [\r\n        {\"role\": \"system\", \"content\": \"You are an assistant who perfectly describes images.\"},\r\n        {\r\n            \"role\": \"user\",\r\n            \"content\": [\r\n                {\"type\" : \"text\", \"text\": \"What's in this image?\"},\r\n                {\"type\": \"image_url\", \"image_url\": {\"url\":" + AnyImageData.text + "\" } }\r\n            ]");

```

Comment:
I see what you do here, it's better to define a function in `Runtime/LLMCharacter.cs` that takes over this part  and can be reused e.g.:
```c#
public async Task<string> ChatWithImage(string query, Uri url, Callback<string> callback = null, EmptyCallback completionCallback = null, bool addToHistory = true)
{
   URLContent urlText  = new URLContent(){url = url.ToString() };
   ImageURLContent urlContent = new ImageURLContent(type="image_url", image_url = urlText)
   TextContent message = new TextContent(){type = "text", text = query};

   string queryWithImage = "[" + JsonUtility.ToJson(message) + "," + JsonUtility.ToJson(urlContent) + "]";
   return await Chat(queryWithImage, callback, completionCallback, addToHistory);
}

public async Task<string> ChatWithImage(string query, Path path, Callback<string> callback = null, EmptyCallback completionCallback = null, bool addToHistory = true)
{
   string queryWithImage = ...
   return await Chat(queryWithImage, callback, completionCallback, addToHistory);
}
```

and inside the `Runtime/LLMInterface.cs`
```c#
    [Serializable]
    public struct TextContent
    {
        public string type;
        public string text;
    }

    [Serializable]
    public struct ImageURLContent
    {
        public string type;
        public URLContent image_url;
    }

    [Serializable]
    public struct URLContent
    {
        public string url;
    }
```

Reply:
Instead of manually defining the "What's in this image?" text, you can use the existing text box in the SimpleInteraction sample.